fv3core/stencils/dyn_core.py

```diff
@@ -26,17 +26,43 @@
 import fv3core.utils.global_config as global_config
 import fv3core.utils.global_constants as constants
 import fv3core.utils.gt4py_utils as utils
+import fv3gfs.util
 import fv3gfs.util as fv3util
-from fv3core.decorators import gtstencil
+from fv3core.decorators import FixedOriginStencil
 from fv3core.stencils.basic_operations import copy_stencil
+from fv3core.utils.grid import axis_offsets
 from fv3core.utils.typing import FloatField, FloatFieldIJ, FloatFieldK
 
 
 HUGE_R = 1.0e40
 
 
+def initialize_data(
+    mfxd: FloatField,
+    mfyd: FloatField,
+    cxd: FloatField,
+    cyd: FloatField,
+    gz: FloatField,
+    pk3: FloatField,
+    diss_estd: FloatField,
+    init_step: bool,
```

Comment:
Should these actually be 2 functions, one of which gets called in an object `__init__`? Note this might need to get called in the `DynamicalCore` init, or wherever these temporaries are allocated.

If not, because "initialize" is used for two different meanings (in this variable name, and in the function name), it would be helpful to document what `init_step` means, and why the initialization is different on the init step. Alternatively, you could update one or both names (`first_timestep: bool`? `zero_data`?).